### PR TITLE
catch errors in zen_workaround

### PR DESCRIPTION
**Pull request by GitMensch**
and ensure handle is closed

side issue of #3251

### CODE REVIEW

File: scripts/zen_workaround.py [hunk 1]
```diff
@@ -53,11 +53,26 @@ def is_secure_boot_enabled():
...
+        os.lseek(msr, MSR, os.SEEK_SET)
+        (val,) = struct.unpack('<q', os.read(msr, 8))
+        return val
+    except PermissionError as e:
+        sys.stderr.write(str(e) + '\n')
+        print("Permission denied on reading from MSR.")
+    except OSError as e:
+        sys.stderr.write(str(e) + '\n')
+        print("General error on reading from MSR.")
+        return 0
```
Comment (rocallahan) [inline scripts/zen_workaround.py:73]: Why is returning 0 a good idea? Shouldn't we kill the script?

Comment (GitMensch) [reply scripts/zen_workaround.py:73]: Returning 0 runs into "not supported" later on and is similar what was done in the other place.
If you want to I can drop the return 0 and break / continue which we have into killing the script after the error (which also may be placed into stderr completely).

**Comment by GitMensch** [comment]
OK, I'll adjust both the old and new code to output to stderr and exit directly.

**Comment by GitMensch** [comment]
Adjusted to exit the script as early as possible when errors occur; added a note to the secure-boot branch as we now never reach the end of the script any more.

**Comment by GitMensch** [comment]
@rocallahan your requested changes were done a year ago, are there more things you want to have adjusted or should this be merged / just closed?

**Comment by rocallahan** [comment]
Merged, sorry for losing this

**Pull request closed without merging**